hydroengine_service/ecopath_functions.py

```diff
@@ -0,0 +1,161 @@
+        return ee.Image(result)
+
+    # Apply the get top layer function to all images
+    hycom_currents = hycom_currents.filterDate(t_start, t_stop)
+    hycom_currents = hycom_currents.map(get_top_layer)
+
+    def get_compatible_glossis_currents(img):
+        result = img.rename(['velocity_u', 'velocity_v'])
+        result = result.resample('bicubic')
+        return result.set('system:time_start', img.get('system:time_start'))
+
+    glossis_currents = glossis_currents.map(get_compatible_glossis_currents)
+
+    def get_map(model, period_range):
+        models = {
+            "HYCOM": hycom_currents,
+            "glossis": glossis_currents
+        }
+
+        selected = models[model]
+        period_range = ee.DateRange(period_range)
+
+        selected = selected.filterDate(
+            period_range.start(),
+            period_range.end()
+        )
+
+        image_count = selected.size()
+        selected = selected.mean()
+        selected = selected.set('model', model)
+        selected = selected.set('tStart', period_range.start())
+        selected = selected.set('tStop', period_range.end())
+        selected = selected.set('imageCount', image_count)
+        selected = selected.set('units', 'm/s')
+        return selected
+    
+    def export_map(image, model, export_path, geometry, scale, crs):  # TODO: model not used
+        task =  ee.batch.Export.image.toCloudStorage(
+            image=image,
+            description=export_path,  # .replace('/', '_').replace(":", '_'),  # Regex for /:/g == .replace(":")?
+            bucket=bucket,
+            fileNamePrefix=export_path,
+            region=geometry,
+            scale=scale,
+            crs=crs 
+        )
+        task.start()
+        return task
+
+    # Now for some annoying stuff, we need to make a list of all properties
+    # so that we can call export with local variables
+
+    def prep_export(period_range):
+        t_start = ee.DateRange(period_range).start()
+        export_path = ee.String("ecopath/").cat(
+            ee.String(model).cat(t_start.format())
+        )
+        im = get_map(model, period_range)
+        im = im.set('path', export_path)
+        im = im.set('tStart',  t_start)
+        im = im.set('system:time_start', t_start.millis())
+        im = im.set('period', period_range)
+        return im
+
+    images = period_ranges.map(prep_export)
+
+    # filter out empty images
+    images = ee.ImageCollection(images)
+    images = images.filter(ee.Filter.gt('imageCount', 0))
+
+    # sort images on path
+    images = images.sort("path")
+    daterange = pd.date_range(start=str_t_start, periods=n_periods, freq=pd.DateOffset(months=1))
+    paths=[f"ecopath_{model}{dr.strftime('%Y-%m-%dT%X').replace(':', '_')}" for dr in daterange]
+    # paths = images.aggregate_array('path')  # TODO: Find a better way to pair paths and images so that .getInfo() is not required
```

Comment:
you need to do 1 get info to get all the "work", because ee.batch.Export.image.toCloudStorage is called clientside